database.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"time"
+
+	"golang.org/x/net/context"
+
+	"google.golang.org/cloud/datastore"
+
+	"github.com/chromium/hstspreload/chromiumpreload"
+)
+
+const (
+	// TODO: Change default to hstspreload and allow a dynamic value.
+	projectId = "hstspreload-mvm"
+	batchSize = 450
+	timeout   = 10 * time.Second
+
+	domainStateKind = "DomainState"
+)
+
+type PreloadStatus string
+
+const (
+	StatusUnknown   = "unknown"
+	StatusPending   = "pending"
+	StatusPreloaded = "preloaded"
+	StatusRejected  = "rejected"
+	StatusRemoved   = "removed"
+)
+
+// DomainState represents the state stored for a domain in the hstspreload
+// submission app database.
+type DomainState struct {
+	// Name is the key in the datastore, so we don't include it as a field
+	// in the stored value.
+	Name chromiumpreload.Domain `datastore:"-" json:"name"`
+	// e.g. StatusPending or StatusPreloaded
+	Status PreloadStatus `json:"status"`
+	// A custom message from the preload list maintainer explaining the
+	// current status of the site (usually to explain a StatusRejected).
+	Message string `datastore:",noindex" json:"message,omitempty"`
+	// The Unix time this domain was last submitted.
+	SubmissionDate time.Time `json:"-"`
+}
+
+// Updates the given domain updates in batches.
+// Writes and flushes updates to w.
+func putStates(updates []DomainState, statusReport func(format string, args ...interface{})) error {
+	if len(updates) == 0 {
+		statusReport("No updates.\n")
```

Comment:
Are you formatting the code with gofmt? (asking because the indendation seems different from google internal styles, but I don't know if it's the default)